core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -659,9 +659,9 @@ private[spark] class Executor(
         val accumUpdates = task.collectAccumulatorUpdates()
         val metricPeaks = metricsPoller.getTaskMetricPeaks(taskId)
         // TODO: do not serialize value twice
-        val directResult = new DirectTaskResult(valueBytes, accumUpdates, metricPeaks)
-        val serializedDirectResult = ser.serialize(directResult)
-        val resultSize = serializedDirectResult.limit()
+        val directResult = new DirectTaskResult(valueByteBuffer, accumUpdates, metricPeaks)
+        val serializedDirectResult = SerializerHelper.serializeToChunkedBuffer(ser, directResult)
```

Comment:
Unlike the earlier invocation of `serializeToChunkedBuffer` (L#599) , here we have a good estimate of the size - something to leverage and minimize the cost of `serializeToChunkedBuffer`

Reply:
do you mean we can explicitly choose the chunk size here to avoid too-small/to-large chunk?

Reply:
Yes, we do know `valueByteBuffer.size` - and we know what the serialization overhead of `DirectTaskResult` is.
Will give us a much tighter bound on the size.